core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -344,7 +382,20 @@ public Pair<Integer, TimeUnit> refresh(RESTClient client) {
             Tasks.foreach(ref)
                 .suppressFailureWhenFinished()
                 .retry(5)
-                .onFailure((task, err) -> LOG.warn("Failed to refresh token", err))
+                .onFailure(
+                    (holder, err) -> {
+                      if (null != credential) {
```

Comment:
I have a test for this behavior but it needs some additional changes to the code (parameterizing the number of retries that are being done so that the default of 5 can be lowered to 1 for testing). @rdblue are we ok adding that test + the code changes for parameterizing the number of retries as a separate PR?

Reply:
Yes, a separate PR is fine.